common/Tests/Framework/src/AITestClass.ts

```diff
@@ -162,6 +193,14 @@ export class AITestClass {
                 let stepIndex = 0;
     
                 const testDone = () => {
+                    if (testInfo.assertNoEvents) {
```

Comment:
New switches (currently false by default) which will automatically fail any test (when true) if any event, hook has not been removed after the test completes.